flink-ml-lib/src/main/java/org/apache/flink/ml/classification/naivebayes/NaiveBayes.java

```diff
@@ -0,0 +1,333 @@
+                .flatMap(new FlattenFunction(
+                        featureColNames,
+                        labelColName
+                ))
+                .keyBy((KeySelector<Tuple4<Object, Integer, Object, Double>, Object>) value -> new Tuple3<>(value.f0, value.f1, value.f2))
+                .window(EndOfStreamWindows.get())
+                .reduce((ReduceFunction<Tuple4<Object, Integer, Object, Double>>) (t0, t1) -> {t0.f3 += t1.f3; return t0; })
+                .keyBy((KeySelector<Tuple4<Object, Integer, Object, Double>, Object>) value -> new Tuple2<>(value.f0, value.f1))
+                .window(EndOfStreamWindows.get())
+                .aggregate(new ValueMapFunction())
+                .keyBy((KeySelector<Tuple4<Object, Integer, Map<Object, Double>, Double>, Object>) value -> value.f0)
+                .window(EndOfStreamWindows.get())
+                .aggregate(new MapArrayFunction(featureColNames.length))
+                .windowAll(EndOfStreamWindows.get())
+                .apply(new GenerateModelFunction(
+                        smoothing,
+                        featureColNames));
+
+        NaiveBayesModel model = new NaiveBayesModel()
+                .setPredictionCol(predictionCol)
+                .setFeatureCols(featureColNames);
+        model.setModelData(
+                tEnv.fromDataStream(naiveBayesModel)
+        );
+        return model;
+    }
+
+    @Override
+    public void save(String path) throws IOException {
+        ReadWriteUtils.saveMetadata(this, path);
+    }
+
+    public static NaiveBayes load(String path) throws IOException {
+        return ReadWriteUtils.loadStageParam(path);
+    }
+
+    @Override
+    public Map<Param<?>, Object> getParamMap() {
+        return paramMap;
+    }
+
+    /**
+     * Function to convert each column into tuples of label, feature column index, feature value, weight.
+     */
+    private static class FlattenFunction implements FlatMapFunction<Row, Tuple4<Object, Integer, Object, Double>> {
+        private final String[] featureColNames;
+        private final String labelColName;
+        private final int featureSize;
+
+        private FlattenFunction(String[] featureColNames, String labelColName) {
+            this.labelColName = labelColName;
+            this.featureColNames = featureColNames;
+            this.featureSize = featureColNames.length;
+        }
+
+        @Override
+        public void flatMap(Row row, Collector<Tuple4<Object, Integer, Object, Double>> collector) {
+            Object label = row.getField(labelColName);
+            if (label == null) {
+                return;
+            }
+
+            for (int i = 0; i < featureSize; i++) {
+                Object feature = row.getField(featureColNames[i]);
+                if (feature == null) {
+                    continue;
+                }
+                collector.collect(new Tuple4<>(label, i, feature, 1.0));
```

Comment:
should we support weighted naive bayes or just remove 1.0 here?

Reply:
We may be able to support weighted training data in Naive Bayes in a later version, but in the current version I believe we only need to achieve the basic functionality of this algorithm.

Although we do not need weighted function here in this version, we still need to add this `1.0` to every training data, because we need to sum up this value and use `weightSum` to calculate the probability of each feature value and each label.